src/queue.cpp

```diff
@@ -757,6 +776,32 @@ cl_int cvk_command_kernel::dispatch_uniform_region_within_vklimits(
     vkCmdDispatch(command_buffer, num_workgroups[0], num_workgroups[1],
                   num_workgroups[2]);
 
+    // If we have a kernel that requires serial execution (i.e. regions are not
+    // executed in parallel with other regions or other kernels) then serialize
+    // the command buffer
+    if (m_kernel->requires_serialized_execution()) {
```

Comment:
I'd have to dive into clspv to confirm but I don't think we need to serialise individual regions.